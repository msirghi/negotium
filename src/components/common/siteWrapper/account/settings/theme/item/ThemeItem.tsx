import { FC } from 'react';
import { useThemeItemStyles } from './styles';
import { ThemeBodySkeleton } from '../skeleton/ThemeBodySkeleton';
import CheckIcon from '@mui/icons-material/Check';
import { Row } from '../../../../../utilities/row/Row';
import { If } from '../../../../../utilities/if/If';
import { SiteTheme } from '../../../../../../../common/constants/types';

type Props = {
  theme: SiteTheme;
  selected?: boolean;
  onClick: (theme: string) => void;
};

export const ThemeItem: FC<Props> = ({ theme, selected, onClick }) => {
  const { color, label, textColor } = theme;
  const classes = useThemeItemStyles({ color, textColor });

  const handleClick = () => {
    onClick(theme.internalKey);
  };

  return (
    <div
      className={classes.container}
      id={`theme-${label}`}
      onClick={handleClick}
    >
      <div>
        <div className={classes.title}>{label}</div>
        <div className={classes.skeleton}>
          <ThemeBodySkeleton />
          <ThemeBodySkeleton />
          <ThemeBodySkeleton />
          <If condition={selected}>
            <Row fullWidth>
              <CheckIcon fontSize={'small'} className={classes.checkIcon} />
            </Row>
          </If>
        </div>
      </div>
    </div>
  );
};
