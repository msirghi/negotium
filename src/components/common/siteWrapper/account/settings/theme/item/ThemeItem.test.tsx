import renderer, { act } from 'react-test-renderer';
import { ThemeItem } from './ThemeItem';
import { mount } from 'enzyme';
import CheckIcon from '@mui/icons-material/Check';

describe('ThemeItem', () => {
  const defaultProps = {
    theme: {
      label: 'red',
      color: 'red',
      internalKey: 'red',
    },
    selected: false,
    onClick: jest.fn(),
  };

  it('should match the snapshot', () => {
    const tree = renderer.create(<ThemeItem {...defaultProps} />);
    expect(tree).toMatchSnapshot();
  });

  it('should render check icon if selected prop is provided', () => {
    const wrapper = mount(<ThemeItem {...defaultProps} selected />);
    expect(wrapper.find(CheckIcon)).toHaveLength(1);
  });

  it('should fire prop method on click', () => {
    const wrapper = mount(<ThemeItem {...defaultProps} />);
    const container = wrapper.find('#theme-red');

    act(() => {
      container.simulate('click');
    });
    expect(defaultProps.onClick).toBeCalled();
  });
});
