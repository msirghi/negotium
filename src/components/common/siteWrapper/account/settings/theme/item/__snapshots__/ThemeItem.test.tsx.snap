// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeItem should match the snapshot 1`] = `
<div
  className="makeStyles-container-1"
  id="theme-red"
  onClick={[Function]}
>
  <div>
    <div
      className="makeStyles-title-2 makeStyles-title-6"
    >
      red
    </div>
    <div
      className="makeStyles-skeleton-3"
    >
      <div
        onClick={[Function]}
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "width": "initial",
          }
        }
      >
        <span
          className="MuiSkeleton-root MuiSkeleton-rectangular css-14wtl70-MuiSkeleton-root"
          style={
            Object {
              "height": 9,
              "width": 10,
            }
          }
        />
        <span
          className="MuiSkeleton-root MuiSkeleton-text makeStyles-textSkeleton-4 css-1l6af8c-MuiSkeleton-root"
          style={
            Object {
              "height": 15,
              "width": "100%",
            }
          }
        />
      </div>
      <div
        onClick={[Function]}
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "width": "initial",
          }
        }
      >
        <span
          className="MuiSkeleton-root MuiSkeleton-rectangular css-14wtl70-MuiSkeleton-root"
          style={
            Object {
              "height": 9,
              "width": 10,
            }
          }
        />
        <span
          className="MuiSkeleton-root MuiSkeleton-text makeStyles-textSkeleton-4 css-1l6af8c-MuiSkeleton-root"
          style={
            Object {
              "height": 15,
              "width": "100%",
            }
          }
        />
      </div>
      <div
        onClick={[Function]}
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "width": "initial",
          }
        }
      >
        <span
          className="MuiSkeleton-root MuiSkeleton-rectangular css-14wtl70-MuiSkeleton-root"
          style={
            Object {
              "height": 9,
              "width": 10,
            }
          }
        />
        <span
          className="MuiSkeleton-root MuiSkeleton-text makeStyles-textSkeleton-4 css-1l6af8c-MuiSkeleton-root"
          style={
            Object {
              "height": 15,
              "width": "100%",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;
